ctests/test_triton_reduction.cpp

```diff
@@ -30,3 +30,29 @@ TEST(reduction_op_test, nonzero) {
 
   EXPECT_TRUE(torch::allclose(out_torch, out_triton, 1e-5, 1e-8));
 }
+
+TEST(reduction_op_test, max_keepdim) {
+    const torch::Device device(torch::kCUDA, 0);
+    for (bool keepdim : {false, true}) {
```

Comment:
For multiple tests, it is recommended to use parametrized tests.

https://google.github.io/googletest/reference/testing.html

Also, there are some examples 
https://github.com/FlagOpen/FlagGems/blob/e6ec851e9b4da5abaf9117384cc28e3fa938df39/ctests/test_triton_blas.cpp#L25C1-L43C1